masonite/view.py

```diff
@@ -51,3 +75,120 @@ def extend(self):
     def share(self, dictionary):
         self.dictionary.update(dictionary)
         return self
+
+    def cache_for(self, time=None, type=None):
+        """
+        Set time and type for cache
+        """
+
+        self.cache_time = time
+        self.cache_type = type
+
+        if self.__is_expired_cache():
+            self.__create_cache_template(self.template)
+        return self
+
+    # --------
+    # Privates
+    # --------
+
+    @property
+    def __get_path_cache(self):
+        """
+        Get path location disk config
+        """
+
+        cache_config = self.container.make('CacheConfig')
+        path = cache_config.DRIVERS['disk']['location'] + "/"
+        return path
+
+    def __create_cache_template(self, template):
+        """
+        Save in the cache the template
+        """
+
+        self.container.make('Cache').store(
+            template + ":" + str(time.time()) + '.html',
+            self.rendered_template
+        )
+
+    def __cached_template_exists(self):
+        """
+        Check if the cache template exists
+        """
+
+        path = self.__get_path_cache
+        find_template = glob.glob(path + "/" + self.template + "*")
+        if find_template:
+            return True
+        return False
+
+    def __is_expired_cache(self):
+        """
+        Check if cache is expired
+        """
+
+        # If is forever
+        if self.cache_time is None and self.cache_time is None:
+            return False
+
+        # By seconds
+        cache_type = self.cache_type.lower()
+        calc = 0
+        if cache_type == "minutes":
```

Comment:
make sure seconds is in here too. Also to make it more developer friendly, sometimes the developer will want to only do `1 minute` or `1 second` or `1 hour` and not `1 minutes`, `1 seconds` or `1 hours` so what we can do is this:

```python
           elif cache_type == "hours" or 'hour':
              calc = 60 * 60
          elif cache_type == "days" or 'day':
              calc = 60 * 60 * 60
          elif cache_type == "months" or 'month':
              calc = 60 * 60 * 60 * 60
          elif cache_type == "years" or 'year':
              calc = 60 * 60 * 60 * 60 * 60
```

for all of them.

Reply:
Perfect!